src/main/java/net/blay09/mods/cookingforblockheads/container/comparator/ComparatorSoL.java

```diff
@@ -40,9 +45,31 @@ public int compare(ItemStack o1, ItemStack o2) {
         } else if (!isFoodSecond) {
             return -1;
         }
-
-        boolean everEatenFirstFood = foodHistory.hasEverEaten(o1);
-        boolean everEatenSecondFood = foodHistory.hasEverEaten(o2);
+        boolean everEatenFirstFood = false, everEatenSecondFood = false;
+        if (!spiceCompat) {
+            everEatenFirstFood = foodHistory.hasEverEaten(o1);
+            everEatenSecondFood = foodHistory.hasEverEaten(o2);
+        } else {
+            // Older versions of Spice of Life don't have full food history, and therefore don't have a hasEverEaten
+            // This has similar, though not identical, behavior for those older versions.
+            try {
+                Method method = foodHistory.getClass().getDeclaredMethod("getHistory");
```

Comment:
This should be cached along with the check for compat; I'd also recommend using a MethodHandle; as an example: https://github.com/GTNewHorizons/Angelica/blob/master/src/main/java/com/gtnewhorizons/angelica/client/renderer/CapturingTessellator.java#L40-L41